### ISSUE TITLE
Injecting environment variables in k8s doesnt work because of the format.

### ISSUE THREAD

**Issue by bozouser** [getsentry/sentry-java#780]
The docs for integrating sentry thru env variables - https://docs.sentry.io/clients/java/config/#configuration-via-the-runtime-environment

```
System Environment Variable option names require that you replace the . with _, capitalize them, and add a SENTRY_ prefix. For example, to enable sampling:
```
these are common options - https://docs.sentry.io/error-reporting/configuration/?platform=aspnetcore#common-options

And this translates to the following - 
1. sample.rate => `SENTRY_SAMPLE_RATE` 
2. max-breadcrumbs => `SENTRY_MAX-BREADCRUMBS`

`2.` fails the deployment.yaml as k8s doesnt allow for `-`

**Comment by metlos** [comment]
Note that java doesn't support the same set of options as the "standardized" platforms. Your link actually points to docs for ASP.NET Core.

The documentation for Java configuration is actually here: https://docs.sentry.io/clients/java/config/

**Comment by bozouser** [comment]
thanks @metlos 2 things:

1. looks like the `java` config doesnt support some of these options like `max-breadcrumbs` thru environment variable injection even though that the most flexible method.

2. The issue stands as you cant inject System variables with `-` in your deployment.yaml.

**Comment by maciejwalkowiak** [comment]
Fixed in https://github.com/getsentry/sentry-java/pull/1145. Will be released in 4.0
